data.tf

```diff
@@ -112,7 +116,8 @@ data "template_file" "userdata" {
       "kubelet_extra_args",
       local.workers_group_defaults["kubelet_extra_args"],
     )
-  }
+    },
+  lookup(var.worker_groups[count.index], "userdata_template_extra_args", {}))
```

Comment:
The default should be `local.workers_group_defaults["userdata_template_extra_args"]` and not `{}` => 

```hcl
lookup(
  var.worker_groups[count.index],
  "userdata_template_extra_args",
  local.workers_group_defaults["userdata_template_extra_args"]
)
```